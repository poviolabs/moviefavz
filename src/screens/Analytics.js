import React from 'react';

import { observer } from 'mobx-react';

import { Layout, Typography } from 'antd';

import { Container, Section } from '../components/layout';
import { Emoji } from '../components/typography';
import { UsageChart } from '../components/ui';

const { Content } = Layout;
const { Title, Paragraph } = Typography;

const Favorites = () => {
  return (
    <Content>
      <Container>
        <Section>
          <Title level={1}>Your App Usage</Title>
          <Paragraph>
            Don&apos;t worry, we do not collect any of your data, this is for
            your eyes only <Emoji emoji="ðŸ‘€" label="pointing finger" />
          </Paragraph>
        </Section>
        <UsageChart />
      </Container>
    </Content>
  );
};

export default observer(Favorites);
