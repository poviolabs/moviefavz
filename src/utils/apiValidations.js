const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
