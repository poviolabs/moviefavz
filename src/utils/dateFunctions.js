import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
