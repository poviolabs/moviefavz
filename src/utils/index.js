export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
