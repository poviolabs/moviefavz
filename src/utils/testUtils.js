import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
