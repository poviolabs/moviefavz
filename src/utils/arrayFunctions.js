import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
