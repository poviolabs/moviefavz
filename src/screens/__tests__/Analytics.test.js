import React from 'react';
import { render } from '../../utils/testUtils';

import Analytics from '../Analytics';

describe('<Analytics />', () => {
  it('Renders the Component', () => {
    const props = {};
    const container = render(<Analytics {...props} />);
    expect(container).toMatchSnapshot();
  });
});
