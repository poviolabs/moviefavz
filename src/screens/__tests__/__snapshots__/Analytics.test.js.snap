// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Analytics /> Renders the Component 1`] = `undefined`;
